services/frontend/domain/usecases/process_manifest_files.go

```diff
@@ -14,152 +10,64 @@ import (
-		go func(manifestFile fs.DirEntry) {
-			var (
-				rawManifestData      []byte
-				unmarshalledManifest *manifest.Manifest
-				processingError      error = nil
-			)
-			defer waitGroup.Done()
-
-			manifestFilepath := filepath.Join(manifestDir, manifestFile.Name())
-
-			defer func() {
-				manifestProcessingResultsChan <- &ManifestProcessingResult{
-					unmarshalledManifest: unmarshalledManifest,
-					rawManifestData:      rawManifestData,
-					manifestFilepath:     manifestFilepath,
-					processingError:      processingError,
-				}
-			}()
-
-			if rawManifestData, processingError = os.ReadFile(manifestFilepath); processingError != nil {
-				return
-			}
-			if processingError = yaml.Unmarshal(rawManifestData, &unmarshalledManifest); processingError != nil {
-				return
-			}
-			processingError = unmarshalledManifest.Validate()
-		}(manifestFile)
+	// Define config
+	config := &pb.Config{
+		Name:             unmarshalledManifest.Name,
+		ManifestFileName: fmt.Sprintf("secret_%s.file_%s", rawManifest.SecretName, rawManifest.FileName),
+		Manifest:         string(rawManifest.Manifest),
 	}
 
-	go func() {
-		waitGroup.Wait()
-		close(manifestProcessingResultsChan)
-	}()
-
-	// Collect processing results of manifest files which were processed successfully
-	for manifestProcessingResult := range manifestProcessingResultsChan {
-		var manifestName string
-		var isConfigRemoved bool
-
-		// Remove the config from configs slice.
-		// After the for loop finishes, the configs variable will contain only those configs which represent
-		// deleted manifest files -> configs which needs to be deleted.
-		configs, isConfigRemoved = removeConfig(configs, manifestProcessingResult.manifestFilepath)
-		// Check for the error first, before referencing any variables.
-		if manifestProcessingResult.processingError != nil {
-			log.Error().Msgf("Skipping over file %v due to processing error : %v", manifestProcessingResult.manifestFilepath, manifestProcessingResult.processingError)
-			continue
-		}
-		manifestName = manifestProcessingResult.unmarshalledManifest.Name
-
-		config := &pb.Config{
-			Name:             manifestName,
-			ManifestFileName: manifestProcessingResult.manifestFilepath,
-			Manifest:         string(manifestProcessingResult.rawManifestData),
-		}
-
-		err = u.ContextBox.SaveConfig(config)
-		if err != nil {
-			log.Error().Msgf("Failed to save config %v due to error : %v", manifestName, err)
-			continue
-		}
-
-		log.Info().Msgf("Details of the manifest file %s has been saved to context-box database", manifestProcessingResult.manifestFilepath)
-
-		// if the config is not in the context-box DB we start to track it.
-		if !isConfigRemoved {
-			for _, k8sCluster := range manifestProcessingResult.unmarshalledManifest.Kubernetes.Clusters {
-				if _, ok := u.inProgress.Load(k8sCluster.Name); !ok {
-					u.inProgress.Store(k8sCluster.Name, config)
-				}
-			}
```

Comment:
Same as with `deleteConfig` func, can we add this code also to the `createConfig` func ? so that the frontend starts tracking the state of the manifest from the moment it was picked up